source/common/secret/secret_manager_impl.h

```diff
@@ -50,7 +50,8 @@ class SecretManagerImpl : public SecretManager {
     findOrCreate(const envoy::api::v2::core::ConfigSource& sds_config_source,
                  const std::string& config_name,
                  Server::Configuration::TransportSocketFactoryContext& secret_provider_context) {
-      const std::string map_key = sds_config_source.SerializeAsString() + config_name;
+      const std::string map_key =
+          std::to_string(MessageUtil::hash(sds_config_source)) + config_name;
```

Comment:
Will a number started config_name potentially conflicts with other hashes? i.e. a config hash `123` with config name `0-abc` conflicts with a config hash `12` with config name `30-abc`, this happens much easier than hash conflict.

Reply:
Potentially, yes, though I'm not sure how well do we guard against that in rest of the codebase.

@quanjielin could you add a separator? e.g.
```
const std::string map_key =
    absl::StrCat(MessageUtil::hash(sds_config_source), ".", config_name);
```
Thanks!

Reply:
Yeah that will be good enough

Reply:
done